spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigWatch.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.config;
+
+import com.microsoft.azure.spring.cloud.config.domain.KeyValueItem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cloud.endpoint.event.RefreshEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.context.SmartLifecycle;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.util.StringUtils;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+public class AzureCloudConfigWatch implements ApplicationEventPublisherAware, SmartLifecycle {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureCloudConfigWatch.class);
+    private final ConfigServiceOperations configOperations;
+    private final LinkedHashMap<String, String> keyNameEtagMap = new LinkedHashMap<>();
```

Comment:
Why use `LinkedHashMap`? Why not use concurrent one?

Reply:
Only one watch will be started, no concurrent is required?

Reply:
If check config hasn't finished in 1 second, and then your next check has already started, then it might be on different thread.

Reply:
Reasonable, updated.